### ISSUE TITLE
SAM deploy issue with Node.js version 

### ISSUE THREAD

**Issue by joozero** [aws-samples/amazon-eventbridge-producer-consumer-example#3]
Since the runtime parameter of nodejs12.x is no longer supported for creating or updating AWS Lambda functions, changing those version to recent one is needed.

**Comment by lonmiller** [comment]
What's strange is sam keeps throwing this error even when I update template.yaml to

```
 atmConsumerCase3Fn:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: atmConsumer/
      Handler: handler.case3Handler
      Runtime: nodejs18.x
```

it still throws the error

```
CREATE_FAILED                    

AWS::Lambda::Function            
atmConsumerCase3Fn    Resource handler returned 
message: "The runtime parameter of nodejs12.x is no 
longer supported for creating or updating AWS lambda 
unctions. We recommend you use the new runtime 
(nodejs18.x) while creating or updating functions. 
```

**Comment by jbesw** [comment]
Hi, I've updated the template to Node. js v18. Please ensure you are running the latest version of SAM CLI before deploying.

**Comment by lonmiller** [comment]
Confirmed I was able to build the project after re-cloning post-fix and updating SAM from 1.104.0 to 1.105.0.
